components/sidebar/sidebar_channel_button_or_link/sidebar_channel_button_or_link.test.jsx

```diff
@@ -156,27 +91,45 @@ describe('component/sidebar/sidebar_channel_button_or_link/SidebarChannelButtonO
         expect(browserHistoryMock.browserHistory.push).not.toBeCalled();
     });
 
-    test('should match snapshot, on group channel with tooltip', () => {
-        const wrapper = shallow(
-            <SidebarChannelButtonOrLink
-                channelType={Constants.GM_CHANNEL}
-                channelId={'test-channel-id'}
-                channelName={'test-channel-name'}
-                channelStatus={'test'}
-                link={'test-link'}
-                rowClass={'test-class'}
-                displayName={'test-channel-name'}
-                handleClose={jest.fn()}
-                hasDraft={false}
-                membersCount={3}
-                showUnreadForMsgs={true}
-                unreadMsgs={2}
-                unreadMentions={6}
-                teammateId={'test-teammate-id'}
-                teammateDeletedAt={1}
-                channelIsArchived={false}
-            />
-        );
-        expect(wrapper).toMatchSnapshot();
+    describe('should properly handle state to show tooltip', () => {
+        for (const testCase of [
+            {
+                name: 'Direct Message',
+                channelType: Constants.DM_CHANNEL,
+            },
+            {
+                name: 'Group Message',
+                channelType: Constants.GM_CHANNEL,
+            },
+            {
+                name: 'Public Channel',
+                channelType: Constants.OPEN_CHANNEL,
+            },
+            {
+                name: 'Private Channel',
+                channelType: Constants.PRIVATE_CHANNEL,
+            },
+        ]) {
+            test(testCase.name, () => {
+                const props = {
+                    ...baseProps,
+                    channelType: testCase.channelType,
+                };
+
+                const wrapper = shallow(
+                    <SidebarChannelButtonOrLink {...props}/>
+                );
+
+                expect(wrapper.find(OverlayTrigger)).toHaveLength(0);
+
+                wrapper.instance().setState({showTooltip: true});
+
+                expect(wrapper.find(OverlayTrigger)).toHaveLength(1);
+                const overlayWrapper = wrapper.find(OverlayTrigger).first();
+                expect(overlayWrapper.prop('overlay').type).toEqual(Tooltip);
+                expect(overlayWrapper.prop('overlay').props.children).toEqual(baseProps.displayName);
+                expect(wrapper).toMatchSnapshot();
```

Comment:
Can we remove saving snapshots here? I find the tests (`expect`) for `OverlayTrigger` and `Tooltip` valuable enough already.

Reply:
Thanks for the great feedback @saturninoabril, appreciated!

Sure, I will do so 🙂